scripting/lua/CMakeLists.txt

```diff
@@ -52,6 +52,6 @@ install(TARGETS vcmiLua DESTINATION ${SCRIPTING_LIB_DIR})
 #manually copy lua dll from vcpkg folder to build directory on windows since vcpkg deps copy feature has flaws, using hardcoded paths based on vcmi windows deps package 1.1 from github
 if(MSVC)
 	if(EXISTS ${LUA_INCLUDE_DIR}/../../bin/lua51.dll)
-		file(COPY ${LUA_INCLUDE_DIR}/../../bin/lua51.dll DESTINATION ${CMAKE_BINARY_DIR}/bin)
+		file(COPY ${LUA_INCLUDE_DIR}/../../bin/lua51.dll DESTINATION ${CMAKE_BINARY_DIR}/bin/${CMAKE_INSTALL_CONFIG_NAME})
```

Comment:
I guess CMAKE_INSTALL_CONFIG_NAME is Release, Debug  or RelWithDebInfo
But it does not works no because LUA_INCLUDE_DIR == ''LUA_INCLUDE_DIR:PATH=D:/Sources/VCMI/vcpkg/installed/x64-windows/include/luajit' and it's not present

Reply:
Could you check if CMAKE_BUILD_TYPE variable does the same? Might be a bit more logical because it is build phase not the installation one

Reply:
BTW this really better to be solved properly without sticking to specific DLL names.

Is there any other projects using lua? What solutions are they using?

Reply:
@ArseniyShestakov, the goal of this PR is to fix build rather than improve it. There might be a better way but it is already merged and as for me it is too late to demand improvements. Thus making the improvement is a question of a separate PR.

Reply:
@nullkiller , @ArseniyShestakov

But I'm confused by the following code here:

if(NOT DEFINED CMAKE_INSTALL_CONFIG_NAME)
if(BUILD_TYPE)
string(REGEX REPLACE "^[^A-Za-z0-9_]+" ""
CMAKE_INSTALL_CONFIG_NAME "${BUILD_TYPE}")
else()
set(CMAKE_INSTALL_CONFIG_NAME "Release")
endif()
message(STATUS "Install configuration: "${CMAKE_INSTALL_CONFIG_NAME}"")
endif()

Why a special case for 'Release' is here? May BUILD_TYPE contain 'Release' value?

Reply:
I think it is some kind of hack to ensure that it is not empty. For instance cpack without parameters supposed to build Release but in NSIS logs I see bin//some.dll which should be bin/release/some.dll. Adding --config Release or whatever cpack wants fixes this issue

Reply:
So, in context of your original question, if BUILD_TYPE may be empty, may be it's more secure to use `${ CMAKE_INSTALL_CONFIG_NAME}` or have I missed something?

Reply:
Need to read about it a bit.

Reply:
If it is some standard thing or not.